connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -165,3 +168,81 @@ export async function fetchZendeskArticlesInCategory({
 
   return response;
 }
+
+export async function fetchZendeskTicketsInBrand({
+  subdomain,
+  accessToken,
+  pageSize,
+  cursor,
+}: {
+  subdomain: string;
+  accessToken: string;
+  pageSize: number;
+  cursor: string | null;
+}): Promise<{
+  tickets: ZendeskFetchedTicket[];
+  meta: { has_more: boolean; after_cursor: string };
+}> {
+  assert(
+    pageSize <= 100,
+    `pageSize must be at most 100 (current value: ${pageSize})`
+  );
+
+  const runFetch = async () =>
+    fetch(
+      `https://${subdomain}.zendesk.com/api/v2/tickets?page[size]=${pageSize}` +
+        (cursor ? `&page[after]=${cursor}` : ""),
+      {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          "Content-Type": "application/json",
+        },
+      }
+    );
+
+  let rawResponse = await runFetch();
+
+  let retryCount = 0;
+  while (await handleZendeskRateLimit(rawResponse)) {
+    rawResponse = await runFetch();
+    retryCount++;
+    if (retryCount >= ZENDESK_RATE_LIMIT_MAX_RETRIES) {
+      logger.info(
+        { response: rawResponse },
+        `[Zendesk] Rate limit hit more than ${ZENDESK_RATE_LIMIT_MAX_RETRIES}, aborting.`
+      );
+      throw new Error(
+        `Zendesk rate limit hit more than ${ZENDESK_RATE_LIMIT_MAX_RETRIES} times, aborting.`
+      );
+    }
+  }
+
+  const text = await rawResponse.text();
+  const response = JSON.parse(text);
+
+  if (!rawResponse.ok) {
+    if (
+      response.type === "error.list" &&
+      response.errors &&
+      response.errors.length > 0
+    ) {
+      const error = response.errors[0];
+      if (error.code === "unauthorized") {
+        throw new ExternalOAuthTokenError();
+      }
+      if (error.code === "not_found") {
+        return { tickets: [], meta: { has_more: false, after_cursor: "" } };
+      }
+    }
+    throw new Error(`Zendesk API error: ${text}`);
+  }
```

Comment:
note: I'll add a generic function to do that in an upcoming PR (I will need this at yet another place + I already have it implemented)

Reply:
noted, don't forget to rebase if that's merged first then to make sure you apply it here